.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,252 @@
+name: Workflow
+
+on: 
+  push:
+    branches:
+      - master
+      - 2-dev
+  pull_request: # This syntax stand for all PRs events
+
+env:
+  NODE_LTS_MAINTENANCE_VERSION: "14"
+  NODE_LTS_ACTIVE_VERSION: "16"
+  NODE_LTS_CURRENT_VERSION: "lts/*"
+  DOCKER_PLATFORMS: "linux/amd64,linux/arm64"
+
+jobs:
+  prepare-matrix:
+    name: Forge Node LTS Matrix
+    runs-on: ubuntu-latest
+    steps:
+      - id: set-matrix
+        run: |
+          echo "::set-output name=matrix::{\"node-version\": [\"$NODE_LTS_MAINTENANCE_VERSION\", \"$NODE_LTS_CURRENT_VERSION\", \"$NODE_LTS_ACTIVE_VERSION\"]}"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+  
+  doc-dead-links:
+    name: Documentation - Check Dead Links
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/install-packages
+      - name: Extract references from context
+        shell: bash
+        id: extract-refs
+        run: |
+          echo "::set-output name=version::$(git describe --abbrev=0 --tags | cut -d. -f 1)"
+          echo "::set-output name=repo::$(echo $GITHUB_REPOSITORY | cut -d/ -f 2)"
+          echo "::set-output name=fw-branch::$(if [ $BASE_BRANCH == master ]; then echo master; else echo develop; fi)"
+      - uses: convictional/trigger-workflow-and-wait@v1.3.0
+        with:
+          owner: kuzzleio
+          repo: documentation
+          github_token: ${{ secrets.ACCESS_TOKEN_CI }}
+          workflow_file_name: dead_links.workflow.yml
+          ref: ${{ steps.extract-refs.outputs.fw-branch }}
+          inputs: '{"repo_name": "${{ steps.extract-refs.outputs.repo }}", "branch": "${{ github.head_ref }}", "version": "${{ steps.extract-refs.outputs.version }}"}'
+
+  error-codes-check:
+    name: Documentation - Error codes check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-packages
+      - run: npm install
+      - name: Test error codes
+        run: ./.ci/scripts/check-error-codes-documentation.sh
+
+  lint:
+    name: Lint - Node.js LTS ${{ matrix.node-version }}
+    runs-on: ubuntu-latest
+    needs: [prepare-matrix]
+    strategy:
+      matrix: 
+        node-version: ${{ fromJson(needs.prepare-matrix.outputs.matrix).node-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-packages
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/cache-node-modules
+        with:
+          NODE_VERSION: ${{ matrix.node-version }}
```

Comment:
Is it normal to have uppercase and lowercase?

Reply:
We could use `node-version` in our custom cache action but this is not the main goal of this PR. I can update it if it is a big concern for you 🙂

Reply:
Updated 😉